feature: `--allow-no-tests` option to pass even if no specifications found

Kingdutch opened the pull request:
In our CI we split out test jobs based on folder. Additionally we run each `.feature` file individually so we can create output test groups. However, due to how we've arranged our tags and our test matrices it could be that Behat is called with a path to a file that has all tests excluded using a tag.

The change in this PR provides an option to mute this error and prevent a CI from failing when no tests were executed deliberately.

Fixes #1389

acoulton reviewed (changes requested):
This feels risky to me - personally I'd want my tooling to explicitly skip combinations of files / tags that were expected not to run any tests. Otherwise there's a chance at some point behat stops finding tests (due to changes in the project or e.g. a behat upgrade) but goes green without running anything.

However, happy to merge it as an opt-in if it's useful to others.

But I find `--allow-empty` a bit unclear. Could it be `--allow-no-tests` instead?

carlos-granados commented on src/Behat/Testwork/Tester/Cli/ExerciseController.php:
```diff
@@ -94,6 +94,9 @@ function ($locator) {
                 'Optional path(s) to execute. Could be:' . PHP_EOL . $locatorsExamples
             )
+            ->addOption('--allow-empty', null, InputOption::VALUE_NONE,
+              'Will not fail if no specification are found.'
```
Should be "if no specifications are found"

Kingdutch commented:
@acoulton @carlos-granados Thanks for your feedback! Took me a bit of time to get to it. I've renamed the flag to `--allow-no-tests`, that's a clearer name. I've also fixed the description to make `specifications` plural.

carlos-granados reviewed (approved):
Thanks @Kingdutch

acoulton reviewed (approved):
Great, thanks for coming back to this @Kingdutch
